### PR TITLE
MediaRanker

### PR DESCRIPTION
**Pull request by daphnegold**
Final Submission

Wow.. look at all those cards in the "done" column. Woohoo! ^_^

https://trello.com/b/NC4gEdo2/media-ranker

### CODE REVIEW

File: app/controllers/media_controller.rb [hunk 1]
```diff
@@ -0,0 +1,101 @@
+class MediaController < ApplicationController
```
Comment (sallyamoore) [inline app/controllers/media_controller.rb:1]: Nice job making a MediaController that works for all types of Media! :tada:

File: app/controllers/media_controller.rb [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  before_action :creator, only: [:new, :edit, :create, :update]
+
+  def index
+    @items = type_class.order(votes: :desc)
+
+    render "index"
+  end
+
+  def show
+    @item = type_class.find(params[:id])
```
Comment (sallyamoore) [inline app/controllers/media_controller.rb:12]: You use this code snippet a few times. Perhaps you could DRY things up by extracting it into a before action.

File: app/controllers/media_controller.rb [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+    case @type
```
Comment (sallyamoore) [inline app/controllers/media_controller.rb:14]: Nice use of a case switch statements!

File: app/controllers/media_controller.rb [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    when "Book" then @special = "Author"
+    when "Movie" then @special = "Director"
+    when "Album" then @special = "Artist"
+    end
+  end
+
+  def type_params
+    params.require(type.downcase).permit(:name, :creator, :description)
+  end
+end
```
Comment (sallyamoore) [inline app/controllers/media_controller.rb:101]: Really nice controller!!! :+1: Your methods are short and clear, and you're doing a great job with single responsibility.

File: app/controllers/welcome_controller.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+class WelcomeController < ApplicationController
+  def index
+    @movies = Movie.order(votes: :desc).first(10)
```
Comment (sallyamoore) [inline app/controllers/welcome_controller.rb:3]: You could DRY this section up by extracting the ordering and selection of top 10 into a model method or scope.

File: app/helpers/albums_helper.rb [hunk 6]
```diff
@@ -0,0 +1,2 @@
+module AlbumsHelper
```
Comment (sallyamoore) [inline app/helpers/albums_helper.rb:1]: You can delete helpers (and other files) you don't wind up using!

**Comment by sallyamoore** [comment]
You did a great job with this project!!! Keep looking for opportunities to DRY things up (although your code is pretty nice and DRY already). You might want to add comments in your code to get in the habit of communicating your intent to other developers who might work with your code in the future. 

Fabulous work! :confetti_ball:

**Comment by daphnegold** [comment]
Thanks for the thorough comments @sallyamoore !

**Pull request closed without merging**